### PR TITLE
README: update badge

### PR DESCRIPTION
**Pull request by scpeters**
# 🦟 Bug fix

Fix Ubuntu CI badge to use focal job instead of bionic

## Summary

Also test macOS CI with `gdal` dependency using https://github.com/osrf/homebrew-simulation/pull/1790

## Checklist
- [X] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [X] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [X] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Review by mjcarroll — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -10,7 +10,7 @@
 Build | Status
 -- | --
 Test coverage | [![codecov](https://codecov.io/gh/ignitionrobotics/ign-common/branch/main/graph/badge.svg)](https://codecov.io/gh/ignitionrobotics/ign-common)
-Ubuntu Bionic | [![Build Status](https://build.osrfoundation.org/buildStatus/icon?job=ignition_common-ci-main-bionic-amd64)](https://build.osrfoundation.org/job/ignition_common-ci-main-bionic-amd64)
+Ubuntu Bionic | [![Build Status](https://build.osrfoundation.org/buildStatus/icon?job=ignition_common-ci-main-focal-amd64)](https://build.osrfoundation.org/job/ignition_common-ci-main-focal-amd64)
```
Comment (jennuine) [inline README.md:13]:
```suggestion
Ubuntu Focal | [![Build Status](https://build.osrfoundation.org/buildStatus/icon?job=ignition_common-ci-main-focal-amd64)](https://build.osrfoundation.org/job/ignition_common-ci-main-focal-amd64)
```

**Comment by scpeters** [comment]
homebrew CI needs https://github.com/ignition-tooling/release-tools/pull/626

**Comment by scpeters** [comment]
I've updated the jenkins homebrew job configuration to use the `scpeters/brew_doctor_ignore_proj7` branch of release-tools

**Comment by scpeters** [comment]
@osrf-jenkins run tests please

**Pull request merged**